### ISSUE TITLE
When initializing react native windows, dependency fails to download

### ISSUE THREAD

**Issue by danielsdesk** [microsoft/react-native-windows#2618]
When following the getting started guide for vnext, upon trying to initialize the project directory like this:
```
react-native windows --template vnext
```
failed output is like this:
```
Reading application name from package.json...
Reading react-native version from node_modules...
Checking for react-native-windows version matching 0.59.*...
Installing react-native-windows@0.59.0-vnext.4...
warning "react-native-windows > react-native-local-cli@1.0.0-alpha.5" has incorrect peer dependency "react-native@^0.57.0".
react-native-windows@0.59.0-vnext.4 successfully installed.
Updating to compatible version of react-native:
    0.59.0-microsoft.1 || https://github.com/Microsoft/react-native/archive/v0.59.0-microsoft.1.tar.gz
error An unexpected error occurred: "https://registry.yarnpkg.com/0.59.0-microsoft.1%20%7C%7C%20https://github.com/Microsoft/react-native/archive/v0.59.0-microsoft.1.tar.gz: Request failed \"405 Method Not Allowed\"".
Command failed: yarn
error An unexpected error occurred: "https://registry.yarnpkg.com/0.59.0-microsoft.1%20%7C%7C%20https://github.com/Microsoft/react-native/archive/v0.59.0-microsoft.1.tar.gz: Request failed \"405 Method Not Allowed\"".
```

**Comment by cloudle** [comment]
@danielsdesk same problem here, seem that we don't have `react-native-windows@0.59.0-vnext.4` yet.

Since latest version of `react-native-windows` is `@0.58.0-rc.2`, I tried to initialize `react-native` at version `@0.58.0-rc.2` and it work:
```
ruui init <projectName> --version react-native@0.58.0-rc.2
```
(we'll want to upgrade it to newer version of `react-native` latter I guess)

**Comment by danielsdesk** [comment]
Thank you, yes doing `react-native init <projectName> --version react-native@0.58.0-rc.2` got me unblocked

**Comment by kmelmon** [comment]
@acoates-ms  it looks like something broke the CLI when we upgraded to 0.59.  I think you've likely already fixed this and verified with the new CLI test you wrote.

**Comment by acoates-ms** [comment]
Yes, this should be fixed by #2641
